sql/2020/05_Media/image_mimetype_ext.sql

```diff
@@ -0,0 +1,26 @@
+#standardSQL
+# images mimetype vs extension
+SELECT
+  client,
+  COUNT(0) AS image_count,
+  ext,
+  mimetype
+FROM
+  (
+  SELECT
+    client,
+    url,
+    mimetype,
+    ext
+  FROM
+    `httparchive.almanac.summary_response_bodies`
```

Comment:
`summary_response_bodies` only contains text responses to save space. So you basically only get SVG images fo this. Use `requests` which has same columns you need but for all requests:

```suggestion
    `httparchive.almanac.requests`
```

Reply:
You are right, it looked a little bit to strange to me that the SVG images were at the top. Now that I have changed to requests I have jpg, png, gif on top, but a lot of low count rows like this

![image](https://user-images.githubusercontent.com/17740806/99983804-0029c500-2db5-11eb-9387-518eafabb171.png)

26k rows. The idea of this, was to figure out if the mimetype is matching the extension, how could I limit this for each of the clients (desktop, mobile)? I can add them all to the spreadsheet, but I think that would be irrelevant.

Reply:
For now, I will get 100 rows for each client and dump them in the spreadsheet in case someone looks for this data.

Reply:
There's always some complete gibberish so let's set a minimum size. The following gives 189 rows. Note I've also removed the grouping sub query as think useful to look at all images a site sends back.

```sql
#standardSQL
# images mimetype vs extension
SELECT
  client,
  ext,
  mimetype,
  COUNT(0) AS ext_mime_image_count,
  SUM(COUNT(0)) OVER (PARTITION BY client) AS total_images,
  SAFE_DIVIDE(COUNT(0), SUM(COUNT(0)) OVER (PARTITION BY client)) AS total_image_pct,
  SAFE_DIVIDE(COUNT(0), SUM(COUNT(0)) OVER (PARTITION BY client, ext)) AS ext_pct,
  SAFE_DIVIDE(COUNT(0), SUM(COUNT(0)) OVER (PARTITION BY client, mimetype)) AS mime_pct
FROM
  `httparchive.almanac.requests`
WHERE
  date = '2020-08-01' AND
  type='image'
GROUP BY
  client, ext, mimetype
HAVING
  ext_mime_image_count > 10000
ORDER BY
  ext_mime_image_count desc, client
```

Reply:
> For now, I will get 100 rows for each client and dump them in the spreadsheet in case someone looks for this data.

Paths crossed. Yeah that's basically what I did. Tried 1000, then when saw that gave 400 rows, I tried 10,000. The other option is to set a LIMIT but difficult to do that for each CLIENT and also I personally prefer setting a hard limit.